AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/AbstractPreferenceListenerHandler.java

```diff
@@ -48,8 +48,8 @@ public void validate(Element element, ElementValidation valid) {
 	}
 
 	@Override
-	protected final JClass getListenerTargetClass() {
-		return getClasses().PREFERENCE;
+	protected final JClass getListenerTargetClass(HasPreferences holder) {
+		return holder.usingSupportV7Preference() ? getClasses().SUPPORT_V7_PREFERENCE : getClasses().PREFERENCE;
```

Comment:
This code : 
```java
holder.usingSupportV7Preference() ? getClasses().SUPPORT_V7_PREFERENCE : getClasses().PREFERENCE;
```
is repeated at least three times in the code. Maybe we could add a method `JClass getPreferenceClass()` in `HasPreferences`.